### PR TITLE
[TOPI] Add GLU operator

### PR DESCRIPTION
**Pull request by nhynes**
This function adds the GLU operator. It's found in language and speech models like [this one](https://github.com/facebookresearch/wav2letter/blob/master/recipes/librispeech/config/conv_glu/network.arch).

On my wishlist for TVM 0.6: make it easier to add new operators. For instance, no having to write a cpp _and_ python version with tests for both. Not needing a bug-prone global export in `topi.cc`. Maybe also automatically generating Relay bindings.

[PyTorch operator, for reference](https://pytorch.org/docs/stable/nn.html?highlight=glu#torch.nn.functional.glu)

### CODE REVIEW

**Comment by merrymercy** [comment]
You should call cpp function in python binding like https://github.com/dmlc/tvm/blob/427e4f1bc8ffae7455b020e7d695994a7ed3e574/topi/python/topi/nn/pooling.py#L42

**Comment by nhynes** [comment]
Thanks for the tip! I wonder if it's worth going the code generation route with AST parsers and the works. IIUC this is akin to what TF does when generating things like the math operators. This shouldn't be too difficult for the C++ side using Clang (just onerous :).

All considered, though, it's great that the Relay op registry doesn't require adding patterns and schedules in Python.

Relatedly, is there a way to ask Relay to schedule `broadcast`? `GLU` doesn't compile under Relay but does for NNVM when registered with 
```python
reg.register_schedule("glu", _fschedule_broadcast)  # <-- this is important, I think
reg.register_pattern("glu", OpPattern.INJECTIVE)
```

And finally, an outlandish proposal: clean up the codebase by removing all TOPI python ops. PyTOPI is useful for prototyping but is confusing when core ops are implemented in multiple places.

**Comment by merrymercy** [comment]
cc @tqchen @jroesch for the simplification of adding new operators.

Relay also requires registrations in python, and you did that in https://github.com/dmlc/tvm/pull/2328/files#diff-b8a5824a5a8f1907448d2588d8cd773cR15 ?

I think your proposal about moving TOPI to cpp is on our roadmap.

**Comment by tqchen** [comment]
Re: automatic wrapper generation.

We had a debate on whether or not we want to automate more when wrapping c++ code from python, I have done so in past related projects. The problem that we quickly encounter, however, are:

- It is hard to generate key-word argument bindings support with exactly the same signature.
   - Doing so require quite a lot of additional meta information
   - Usually, the best effort we can do still leaves things like ```myfunc(*args, **kwargs)```
- It is hard to automatically expose clear doc-strings.
- It usually confuses developer because they can not find the auto-generated function.

In summary, the automatically generated functions are OK, but not the most user-friendly one with the right keyword argument and positional support and clear docs. So we made a decision to not automate that much to
- Always only exposes function with positional arguments (PackedFunc) to the frontend, this simplifies the complexity and indirection in FFI 
- Always create a wrapper function for user-facing APIs, with clear doc-strings.

We believe that because it is usually one-liner to add such indirection and a lot of effort was spent in making clear docs, this simplified approach will not bring too much additional burden to the developers but helps our users by a lot.

**Comment by jroesch** [comment]
@nhynes the broadcast schedule is just a synonym for the injective scheduling (https://github.com/dmlc/tvm/blob/master/nnvm/python/nnvm/top/tensor.py#L42), if you are having an issue compiling with Relay could you open an issue with the right details to reproduce? I would be happy to chase down the bug for you.

**Comment by nhynes** [comment]
@jroesch thanks, it's fixed now due to @merrymercy 's observation (I'd originally made it eltwise). This PR is actually ready for review now.

**Comment by tqchen** [comment]
Given that glu can be implemented with split+ sigmoid, would it be easier to just implement an importer that converts the glu into the composed ops?
